udata/tests/api/test_organizations_api.py

```diff
@@ -682,22 +684,63 @@ def test_list(self):
     def test_create(self):
         data = self.factory.attributes()
         with self.api_user():
-            response = self.post(
-                url_for('api.organization_badges', org=self.organization),
-                data)
+            with self.assert_emit(on_badge_added):
```

Comment:
I think you can wrap this in a single context manager (if it pass the 80 chars constraint)